### ISSUE TITLE
'GridSearch' object has no attribute 'predictors_'

### ISSUE THREAD

**Issue by shoresh57** [fairlearn/fairlearn#621]
AttributeError                            Traceback (most recent call last)
<ipython-input-502-cb592bed5a23> in <module>
      2           sensitive_features=A_train)
      3 
----> 4 predictors = sweep.predictors_

AttributeError: 'GridSearch' object has no attribute 'predictors_'

**Comment by riedgar-ms** [comment]
This sounds like you have `v0.4.6` of Fairlearn installed from `pip`, but are following instructions for `master` ?

**Comment by shoresh57** [comment]
Yes , I exactly follow instruction from https://github.com/fairlearn/fairlearn/blob/master/notebooks/Binary%20Classification%20with%20the%20UCI%20Credit-card%20Default%20Dataset.ipynb

Pip install fairlearn 0.4.6

I get two error :

1-  from fairlearn.metrics import MetricFrame

2-  sweep_preds = [predictor.predict(df_test) for predictor in sweep.predictors_]




Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by riedgar-ms** [comment]
We've had quite a few breaking changes unfortunately. We're hoping to get `v0.5.0` out this week, but in the mean time, you can clone the Fairlearn repository and `pip install .` on it, which will give you a match between the library and examples.

**Comment by shoresh57** [comment]
you mean :
git clone git@github.com:fairlearn/fairlearn.git

pip install -r requirements.txt

**Comment by riedgar-ms** [comment]
Once you've cloned the repo, you can just run `pip install .` from its root. The requirements will be automatically installed (if you don't have them already)

**Comment by shoresh57** [comment]
Thanks

**Comment by riedgar-ms** [comment]
Are you unblocked now?

**Comment by shoresh57** [comment]
no , clone of rep also didnt help

**Comment by shoresh57** [comment]
need to clone repo and install

**Comment by shoresh57** [comment]
if we clone the git repo then we face another problem in widge and dashboard can not be seen

**Comment by riedgar-ms** [comment]
Did you `pip install .` or `pip install -e .` - I believe that you have to do the former at least once.

**Comment by romanlutz** [comment]
Shouldn't make a difference with the flask server anymore as long as you do install it. Where are you running this?

**Comment by shoresh57** [comment]
I used Azure ml notebook and VScode , the problem pip install . has some clash with azure machine learning environment ,and probably we need to create new environment which it would be great that you add this procedure in setup and installation documentation with git clone

**Comment by riedgar-ms** [comment]
That sounds exactly like the problem @romanlutz is investigating now.

Are you able to run the notebook on your local machine?

**Comment by shoresh57** [comment]
@riedgar-ms @romanlutz  if you install with pip fairlearn then local machine can see dashboard but you face problem with 'predictors_ and metricsframe and if you clon the rep and pip install . then the problem of 'predictors_ and metricsframe will be solved but then you get error to import dashboard since they are not match..

**Comment by riedgar-ms** [comment]
I'm not quite clear on what you're doing. I've just been able to do the following:

1. Create a new conda environment (or whatever you prefer)
1. Run `pip install git+https://github.com/fairlearn/fairlearn.git` (on Windows, I also had to run `pip install --upgrade numpy=1.19.3` thanks to a disagreement between Windows and numpy 1.19.4)
1. Run the [census GridSearch notebook](https://fairlearn.github.io/master/auto_examples/plot_grid_search_census.html) and see the dashboard

Are those steps not working for you?

**Comment by romanlutz** [comment]
We rolled the dashboard back to the previous version before the v0.5.0 release so this problem shouldn't occur anymore. We validated that AML and local notebooks work with the dashboard in v0.5.0. If that's not working for you please open a new issue (since this one is for different problem).

The previous issue is addressed by versioned documentation. You should find a version selector on the webpage in the top left corner.
